### PR TITLE
link back to terra based on url query

### PR DESCRIPTION
**Pull request by zarsky-broad**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [ ] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

Twinned with https://github.com/broadinstitute/firecloud-develop/pull/1414
See https://github.com/DataBiosphere/saturn-ui/issues/770 and https://github.com/DataBiosphere/saturn-ui/issues/771

### CODE REVIEW

**Comment by davidangb** [comment]
Code looks fine, but I'm rather confused about the functionality. Is there a description anywhere of what this is trying to achieve? Acceptance criteria?

File: src/cljs/main/broadfcui/common.cljs [hunk 1]
```diff
@@ -245,3 +245,5 @@
 ;; scopes live here instead of in auth.cljs as a quick fix to avoid circular dependencies
 (def login-scopes ["email" "profile"])
 (def storage-scopes (conj login-scopes "https://www.googleapis.com/auth/devstorage.read_only"))
+
+(def has-terra-return? (string/includes? js/window.location.search "return=terra"))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/common.cljs]: Does this need to be a function? Seems like this should be constant in normal usage, but I'm imagining some power user manually editing their URL bar.

**Review by MattPutnam — APPROVED** [review]
👍 with one question

File: src/cljs/main/broadfcui/common.cljs (hunk 1 above)
Comment (zarsky-broad) [reply src/cljs/main/broadfcui/common.cljs]: Yeah, just in case of that and anything else, I wanted it to be evaluated dynamically.

Comment (MattPutnam) [reply src/cljs/main/broadfcui/common.cljs]: What I mean is, right now it won't do that.

**Comment by katevoss** [comment]
@davidangb The full Acceptance Criteria and reasoning behind this PR is recorded in https://github.com/DataBiosphere/saturn-ui/issues/771 and https://github.com/DataBiosphere/saturn-ui/issues/770.

_TLDR:_ when a user in Saturn navigates to the FireCloud Data Repository or FireCloud Methods Repository, we want them to see the Data repo and Methods Repo but only a minimal view so they don't navigate around FireCloud. Instead, we want them to export the data and methods back to Saturn.

File: src/cljs/main/broadfcui/common.cljs (hunk 1 above)
Comment (zarsky-broad) [reply src/cljs/main/broadfcui/common.cljs]: Ah, good catch, I think I got distracted on flipping this to a defn.

**Review by davidangb — APPROVED** [review]
:+1: thanks for the story links - makes sense! There are still some corner-case ways to break out of the Terra-return ... for instance, if the user's account is linked with era commons, but is expired, the user will be offered a banner asking them to re-link; in the redirects, the `?return=terra` gets lost. But I think you probably don't want to bother with these infrequent corner cases.

**Comment by zarsky-broad** [comment]
Indeed, we encountered that, but since the user can go back we figured it wasn't an issue.

**Comment by katevoss** [comment]
@zarsky-broad @davidangb @breilly2 could you look at the Jenkins test errors? 

from https://fc-jenkins.dsp-techops.broadinstitute.org/job/firecloud-ui-fiab-test-runner/8055/console:
```
[info] *** 1 TEST FAILED ***
[error] Failed: Total 123, Failed 1, Errors 0, Passed 122, Ignored 7
[error] Failed tests:
[error] 	org.broadinstitute.dsde.firecloud.test.security.AuthDomainMultiGroupsSpec
[error] (Test / testOnly) sbt.TestsFailedException: Tests unsuccessful
```

**Comment by katevoss** [comment]
jenkins retest

**Comment by davidangb** [comment]
@katevoss I've not seen that particular error before, but it looks unrelated to your changes, and also looks like it passed on a re-run. I'm :+1: to merge.

**Comment by katevoss** [comment]
@elizzhao this is merged! You should now be unblocked to finish https://github.com/DataBiosphere/saturn-ui/issues/770

**Pull request merged**